ydb/core/tx/schemeshard/schemeshard_utils.cpp

```diff
@@ -348,39 +351,47 @@ void SetImplTablePartitionConfig(
     *tableDescription.MutablePartitionConfig() = PartitionConfigForIndexes(baseTablePartitionConfig, indexTableDesc);
 }
 
- void FillIndexImplTableColumns(
-    const THashMap<ui32, struct NSchemeShard::TTableInfo::TColumn>& baseTableColumns,
-    const NTableIndex::TTableColumns& implTableColumns,
+void FillIndexImplTableColumns(
+    const auto& baseTableColumns,
+    const TTableColumns& implTableColumns,
     NKikimrSchemeOp::TTableDescription& implTableDesc)
 {
-    //Columns and KeyColumnNames order is really important
-    //the order of implTableColumns.Keys is the right one
+    const auto was = implTableDesc.ColumnsSize();
 
     THashMap<TString, ui32> implKeyToImplColumn;
     for (ui32 keyId = 0; keyId < implTableColumns.Keys.size(); ++keyId) {
         implKeyToImplColumn[implTableColumns.Keys[keyId]] = keyId;
     }
 
-    for (auto& iter: baseTableColumns) {
-        const NSchemeShard::TTableInfo::TColumn& column = iter.second;
-        if (column.IsDropped()) {
-            continue;
+    for (ui32 i = Max<ui32>() / 2; auto& iter: baseTableColumns) {
+        NKikimrSchemeOp::TColumnDescription* item = nullptr;
```

Comment:
`item` is not a good name

Reply:
it was

Reply:
You have done a total refactoring of this function. So the main branch wrong name is no longer matters.
Please, rename to `column`.

Reply:
column is already used